sysdig/data_source_sysdig_secure_onboarding.go

```diff
@@ -282,6 +282,72 @@ func dataSourceSysdigSecureAgentlessScanningAssetsRead(ctx context.Context, d *s
 	return nil
 }
 
+func dataSourceSysdigSecureCloudIngestionAssets() *schema.Resource {
+	timeout := 5 * time.Minute
+
+	return &schema.Resource{
+		ReadContext: dataSourceSysdigSecureCloudIngestionAssetsRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(timeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"aws": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"gcp_routing_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"gcp_metadata": {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
```

Comment:
Any reason we use a map for AWS, but individual strings for GCP?

Reply:
So we get a map of values from the API, both for AWS and GCP. Original change was using a map for both.

In GCP case however, the map itself is a mix of string and another map. To be able to read GCP's values, we would have to do nested map in the schema then. That was making it complicated to parse and do schema conversions. Hence this separation of routing key and metadata looks cleaner and TF can directly read them.

We can change AWS to use string too for the ARN. But incase we add more ingestion related assets in future, this would need to change.

Reply:
Makes sense, thanks for the explanation!

Reply:
Ofcourse, np!